### PR TITLE
Release candidate

### PR DESCRIPTION
**Pull request by Sandy247**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.idea/*
```
Comment (amoghmargoor) [inline .gitignore]: Can you remove .gitignore file from checkin ?

File: src/main/java/com/qubole/qds/sdk/java/entities/SubCommandsDeserializer.java [hunk 2]
```diff
@@ -51,21 +51,21 @@ public SubCommands deserialize(JsonParser jp, DeserializationContext ctxt)
...
-                node.getFields();
+                node.fields();
         while (elementsIterator.hasNext())
         {
             Map.Entry<String, JsonNode> element = elementsIterator.next();
             String name = element.getKey();
             JsonNode val = element.getValue();
             if (name.equalsIgnoreCase("sub_commands")) {
-                Command[] subCommands = mapper.readValue(val, Command[].class);
+                Command[] subCommands = mapper.readValue(jp, Command[].class);
```
Comment (amoghmargoor) [inline src/main/java/com/qubole/qds/sdk/java/entities/SubCommandsDeserializer.java]: shouldn't this be val ?

File: .gitignore (hunk 1 above)
Comment (Sandy247) [reply .gitignore]: Done

File: src/main/java/com/qubole/qds/sdk/java/entities/SubCommandsDeserializer.java (hunk 2 above)
Comment (Sandy247) [reply src/main/java/com/qubole/qds/sdk/java/entities/SubCommandsDeserializer.java]: The new API definition for readValue() requires the JsonParser object and not the JsonNode object.

Comment (amoghmargoor) [reply src/main/java/com/qubole/qds/sdk/java/entities/SubCommandsDeserializer.java]: ok.

**Pull request merged**